src/components/QueuedTransactionList/index.tsx

```diff
@@ -0,0 +1,190 @@
+import { Identicon, Switch, Text } from '@gnosis.pm/safe-react-components'
+import { getTransactionQueue, TransactionSummary } from '@gnosis.pm/safe-react-gateway-sdk'
+import { FormControlLabel, List } from '@material-ui/core'
+import ChevronRightIcon from '@material-ui/icons/ChevronRight'
+import { ReactElement, useEffect, useState } from 'react'
+import { useSelector } from 'react-redux'
+import { Link } from 'react-router-dom'
+
+import { CircleDot } from 'src/components/AppLayout/Header/components/CircleDot'
+import PrefixedEthHashInfo from 'src/components/PrefixedEthHashInfo'
+import { getChainById } from 'src/config'
+import { ChainId } from 'src/config/chain.d'
+import useLocalSafes from 'src/logic/safe/hooks/useLocalSafes'
+import useOwnerSafes from 'src/logic/safe/hooks/useOwnerSafes'
+import {
+  isMultisigExecutionInfo,
+  isTransactionSummary,
+  LocalTransactionStatus,
+} from 'src/logic/safe/store/models/types/gateway.d'
+import { userAccountSelector } from 'src/logic/wallets/store/selectors'
+import { generateSafeRoute, SAFE_ROUTES } from 'src/routes/routes'
+import { grey400 } from 'src/theme/variables'
+import { checksumAddress } from 'src/utils/checksumAddress'
+import { GATEWAY_URL } from 'src/utils/constants'
+
+import styled from 'styled-components'
+
+const getTxsAwaitingConfirmationByChainId = async (
+  chainId: string,
+  safeAddress: string,
+  account: string,
+  ownedSafes = false,
+): Promise<TransactionSummary[]> => {
+  const { results } = await getTransactionQueue(GATEWAY_URL, chainId, checksumAddress(safeAddress))
+  return results.reduce((acc, cur) => {
+    if (
+      isTransactionSummary(cur) &&
+      isMultisigExecutionInfo(cur.transaction.executionInfo) &&
+      cur.transaction.txStatus === LocalTransactionStatus.AWAITING_CONFIRMATIONS &&
+      (ownedSafes ? cur.transaction.executionInfo.missingSigners?.some(({ value }) => value === account) : true)
```

Comment:
I would suggest moving this logic outside of the `if`. It's a very complex condition to begin with.